### PR TITLE
CNFT1-2431 Patient search criteria: Basic Info

### PR DESCRIPTION
**Pull request by jlee-enquizit**
## Description

This is Basic Information fields in Patient Search. This includes the Accordion component.

## Tickets

* [CNFT1-2431](https://cdc-nbs.atlassian.net/browse/CNFT1-2431)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2431]: https://cdc-nbs.atlassian.net/browse/CNFT1-2431?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/apps/search/patient/PatientCriteria/BasicInformation.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import { Input } from 'components/FormInputs/Input';
+import { validNameRule } from 'validation/entry';
+import { DatePickerInput } from 'components/FormInputs/DatePickerInput';
+import { SelectInput } from 'components/FormInputs/SelectInput';
+import { FormGroup, Label, Checkbox, ErrorMessage } from '@trussworks/react-uswds';
+import { RecordStatus } from 'generated/graphql/schema';
+import styles from './basic-information.module.scss';
+
+type Props = {
+    control: any;
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/search/patient/PatientCriteria/BasicInformation.tsx:12]: Issue: `any`

File: apps/modernization-ui/src/components/Accordion/Accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    title?: string;
+    children: ReactNode;
+    open?: boolean;
+};
+
+export const Accordion = ({ title, children, open }: Props) => {
+    return (
+        <details className={styles.accordian} {...(open && { open: true })}>
+            <summary>
+                <h3>{title}</h3> <span></span>
```
Comment (rkrusselenq) [inline apps/modernization-ui/src/components/Accordion/Accordion.tsx:14]: you have a empty span?

**Review by rkrusselenq — APPROVED** [review]

File: apps/modernization-ui/src/components/Accordion/Accordion.tsx (hunk 2 above)
Comment (jlee-enquizit) [reply apps/modernization-ui/src/components/Accordion/Accordion.tsx:14]: the span is the icon

File: apps/modernization-ui/src/components/Accordion/accordion.module.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        input {
+            width: calc(100% - 1rem);
+            height: 1.5rem;
+            margin-bottom: 0.5rem;
+        }
+        select {
+            width: calc(100% - 2.5rem);
+            height: 1.5rem;
+            margin-bottom: 0.5rem;
+        }
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/components/Accordion/accordion.module.scss:44]: (non-blocker) ideally the accordion would not have special rules for content.  I wonder what is causing the `input` or `select` elements to render so tall.

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]

File: apps/modernization-ui/src/apps/search/patient/PatientSearch.tsx [hunk 4]
```diff
@@ -7,11 +7,18 @@ import { SearchLayout, SearchResultList } from 'apps/search/layout';
...
-    const { handleSubmit, reset: resetForm } = useForm<PatientCriteriaEntry, Partial<PatientCriteriaEntry>>({
+    const {
+        control,
+        trigger,
+        handleSubmit,
+        reset: resetForm
+    } = useForm<PatientCriteriaEntry, Partial<PatientCriteriaEntry>>({
         defaultValues: initial,
-        mode: 'onBlur'
+        mode: 'onChange'
```
[the remark this answers is not in the document]
Comment (jlee-enquizit) [reply apps/modernization-ui/src/apps/search/patient/PatientSearch.tsx:21]: onChange seems the only way the CheckboxGroup would validate.
Is there another way?

File: apps/modernization-ui/src/components/Accordion/accordion.module.scss (hunk 3 above)
Comment (jlee-enquizit) [reply apps/modernization-ui/src/components/Accordion/accordion.module.scss:44]: USWDS Inputs and Select have all kinds of padding and margins -- a lot of times they're placeholders for error messages.

File: apps/modernization-ui/src/apps/search/patient/PatientCriteria/PatientCriteria.tsx [hunk 5]
```diff
@@ -1,49 +1,26 @@
...
-import {  } from 'types/patients';
-import { transformObject } from './transformer';
+import styles from './patient-criteria.module.scss';
 
-type Props = {
-    handleSubmission: (data: PersonFilter) => void;
-    personFilter: PersonFilter | undefined;
-    clearAll: () => void;
+type PatientCriteriaFormProps = {
+    control: Control<PatientCriteriaEntry>;
```
[the remark this answers is not in the document]
Comment (jlee-enquizit) [reply apps/modernization-ui/src/apps/search/patient/PatientCriteria/PatientCriteria.tsx]: Done!

File: apps/modernization-ui/src/apps/search/patient/PatientSearch.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mpeels) [reply apps/modernization-ui/src/apps/search/patient/PatientSearch.tsx:15]: May have to update the `CheckboxGroup` to accept `onBlur` from the controller and pass it down to each checkbox.

[the remark this answers is not in the document]
Comment (jlee-enquizit) [reply apps/modernization-ui/src/apps/search/patient/PatientSearch.tsx:15]: I'll try that...

[the remark this answers is not in the document]
Comment (jlee-enquizit) [reply apps/modernization-ui/src/apps/search/patient/PatientSearch.tsx]: Added onBlur to CheckboxGroup

File: apps/modernization-ui/src/design-system/checkbox/CheckboxGroup.tsx [hunk 6]
```diff
@@ -34,6 +46,7 @@ export const CheckboxGroup = ({ name, label, options, value = [], onChange, disa
     return (
         <fieldset className={classNames(styles.checkboxGroup, className)}>
             <legend>{label}</legend>
+            {error ? <ErrorMessage>{error}</ErrorMessage> : null}
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/design-system/checkbox/CheckboxGroup.tsx:49]: praise:  Thanks for adding this!

**Review by adamloup-enquizit — APPROVED** [review]
👍 👍  great job

**Review by mpeels — APPROVED** [review]

**Pull request merged**